### PR TITLE
Refactor / Auth Module

### PR DESCRIPTION
**Pull request by nahuelRosas**
### Summary

This PR includes several substantial updates to the authentication module of the NestJS project. The main changes are:

1. **JWT Strategy**: A new `JwtStrategy` has been implemented for JWT authentication, with specific handling for automated testing.

2. **Auth Module Refactor**: Significant updates to the `AuthModule`, including the addition of new providers and the removal of deprecated services and modules.

3. **Auth Controller Updates**: Expanded `AuthController` with new endpoints for user registration, login, password management, and session refresh.

4. **File Removals**: Several outdated files and components have been deleted to clean up the module.

#### Modifications

1. **`src/modules/auth/application/strategy/jwt.strategy.ts`**
   - Added a new strategy for JWT authentication, with separate configurations for testing and production environments.

2. **`src/modules/auth/auth.module.ts`**
   - Refactored imports and providers, including the new `JwtStrategy`.
   - Corrected module setup and removed deprecated components.

3. **`src/modules/auth/domain/auth_type.enum.ts`**
   - Introduced `AuthType` enum for defining authentication types.

4. **`src/modules/auth/interface/auth.controller.ts`**
   - Updated controller to include new routes and methods for managing user accounts and sessions.

5. **Deleted Files**
   - Removed various outdated and unused files, including:
     - `role.enum.ts`
     - `user.domain.ts`
     - `aws.cognito.service.ts`
     - `auth.decorators.ts`
     - `admin-role.guard.ts`
     - `auth-team.guard.ts`
     - `owner-role.guard.ts`
     - `jwt.strategy.ts`
     - `user.schema.ts`
     - `user.typeorm.repository.ts`
     - `auth.e2e.spec.ts`

6. **`src/test/test.constants.ts`**
   - Added a constant for the JWT secret used in automated tests.

### Files Changed

- **Lines Added**: 1209
- **Lines Deleted**: 719

### Summary of Changes

- **New Additions**: Implemented `JwtStrategy`, updated `AuthController`, and added new routes and methods.
- **Deletions**: Removed a significant number of outdated files and components to streamline the authentication module.

### CODE REVIEW

File: src/modules/auth/application/interface/auth.controller.interface.ts
Comment (anarod13) [inline src/modules/auth/application/interface/auth.controller.interface.ts:1]: As per the architecture we're following in this project is not necessary to implement an interface for the app services or its controllers. We'll use them only for external services or 3rd parties repositories. Ex: An interface for a cognito service would be useful cause it allow us to keep our code decoupled and independent from external services, but one for a service with our own logic service wouldn't be necessary, as it's core part for our app. Let's remove this one and the service one please

File: src/modules/auth/application/interface/jwt.strategy.interface.ts
Comment (anarod13) [inline src/modules/auth/application/interface/jwt.strategy.interface.ts:1]: Same here

**Review by anarod13 — APPROVED** [review]

File: src/modules/auth/application/interface/auth.controller.interface.ts
Comment (nahuelRosas) [reply src/modules/auth/application/interface/auth.controller.interface.ts:1]: Fixed

File: src/modules/auth/application/interface/jwt.strategy.interface.ts
Comment (nahuelRosas) [reply src/modules/auth/application/interface/jwt.strategy.interface.ts:1]: Fixed

**Pull request merged**